### PR TITLE
[MSYS-722] Escape forward slash for elasticsearch query only

### PR DESCRIPTION
**Pull request by piyushawasthi**
**Issue:**
ZD 17021 - - &lt;Knife search not working with forward slash &gt;

**Testcase for Elasticsearch:**
```
msys@msys-Lenovo-E41-80:~/chef-repo/chef$ bundle exec knife search dh_bag "id:dh/*"
1 items found

chef_type: data_bag_item
comment:   Dheeraj bag
data_bag:  dh_bag
gid:       ops
id:        dh
path:      /test/path/dh
shell:     /bin/bash
uid:       1005

msys@msys-Lenovo-E41-80:~/chef-repo/chef$ bundle exec knife search dh_bag "id:dh//*"
1 items found

chef_type: data_bag_item
comment:   Dheeraj bag
data_bag:  dh_bag
gid:       ops
id:        dh
path:      /test/path/dh
shell:     /bin/bash
uid:       1005

msys@msys-Lenovo-E41-80:~/chef-repo/chef$ bundle exec knife search dh_bag "id:dh///*"
1 items found

chef_type: data_bag_item
comment:   Dheeraj bag
data_bag:  dh_bag
gid:       ops
id:        dh
path:      /test/path/dh
shell:     /bin/bash
uid:       1005

msys@msys-Lenovo-E41-80:~/chef-repo/chef$ bundle exec knife search dh_bag "id:dh//////////*"
1 items found

chef_type: data_bag_item
comment:   Dheeraj bag
data_bag:  dh_bag
gid:       ops
id:        dh
path:      /test/path/dh
shell:     /bin/bash
uid:       1005
```

**Testcase for Solr:**
```
msys@msys-Lenovo-E41-80:~/chef-repo/chef$ bundle exec knife search ammy_bg "path:chef*"
1 items found

chef_type: data_bag_item
data_bag:  ammy_bg
id:        ammy
path:      chef/test/check

msys@msys-Lenovo-E41-80:~/chef-repo/chef$ bundle exec knife search ammy_bg "path:chef/*"
1 items found

chef_type: data_bag_item
data_bag:  ammy_bg
id:        ammy
path:      chef/test/check

msys@msys-Lenovo-E41-80:~/chef-repo/chef$ bundle exec knife search ammy_bg "path:chef//*"
0 items found

msys@msys-Lenovo-E41-80:~/chef-repo/chef$ bundle exec knife search ammy_bg "path:chef\//*"
0 items found

msys@msys-Lenovo-E41-80:~/chef-repo/chef$ bundle exec knife search ammy_bg "path:chef\\//*"
0 items found

msys@msys-Lenovo-E41-80:~/chef-repo/chef$ bundle exec knife search ammy_bg "path:chef/////*"
0 items found
```

Signed-off-by: piyushawasthi <piyush.awasthi@msystechnologies.com>

### CODE REVIEW

**Comment by piyushawasthi** [comment]
Hi @PrajaktaPurohit, @stevendanna - If you have time please have a look this PR and let me know your thought about it. I am just escaping forward_slash here in elasticsearch query only due to I want those changes should not affect anything else.

**Comment by PrajaktaPurohit** [comment]
Thank you very much for working through this!

Does the search with a '/' still work with solr as expected?

**Comment by piyushawasthi** [comment]
@PrajaktaPurohit  - added some test for solr as well.

**Comment by PrajaktaPurohit** [comment]
@piyushawasthi sorry I see you have mentioned that you have added some tests for solr, But I do not see them. Also this branch probably needs to be rebased with the base branch.

File: src/oc_erchef/apps/chef_index/src/chef_lucene.peg [hunk 1]
```diff
@@ -65,7 +65,7 @@ start_letter <- [a-zA-Z0-9_.*/] / '\\' special_char ~;
 
 space <- [ \t]+ ~;
 
-special_char <- '[' / ']' / '\\' / '"' / [!(){}^~*?:] ~;
+special_char <- '[' / ']' / '\\' / '/' / '"' / [!(){}^~*?:] ~;
```
[the remark this answers is not in the document]
Comment (piyushawasthi) [reply src/oc_erchef/apps/chef_index/src/chef_lucene.peg]: Sure @PrajaktaPurohit, I'll add some test cases for that.

**Pull request closed without merging**